File: HackerEarth/Sep/Sep_Circuits/Rescuer.cpp
#include <bits/stdc++.h>
const int mod = 1e9 + 7;
using namespace std;
double X1, X2, Y1, Y2, v1, v2, Y12, Y22;
int t;
double eps = 1;

double findMin( double low,  double high) {
	double res = 1e18;
	while (low <= high) {
		double mid1 = low + (high - low) / 3.0;
		double mid2 = high - (high - low)  / 3.0;
		double t1 = (sqrt((X1 - mid1) * (X1 - mid1) + ( -Y1) * ( -Y1)) / v1) + (sqrt((X2 - mid1) * (X2 - mid1) + ( -Y2) * ( -Y2)) / v2);
		double t2 = (sqrt((X1 - mid2) * (X1 - mid2) + ( -Y1) * ( -Y1)) / v1) + (sqrt((X2 - mid2) * (X2 - mid2) + ( -Y2) * ( -Y2)) / v2);

		// cout << fixed << setprecision(7) << mid << " " << midpv << " " << midv << " " << midnv << '\n';
		if (low == high) {
			double t3 = (sqrt((X1 - low) * (X1 - low) + ( -Y1) * ( -Y1)) / v1) + (sqrt((X2 - low) * (X2 - low) + ( -Y2) * ( -Y2)) / v2);
			res = min(res, t3);
			break;
		}
		if (t1 < t2) {
			high = mid2 - 1;
			res = min(res, t1);
		}
		else if (t1 > t2) {
			res = min(res, t2);
			low = mid1 + 1;
		} else  {
			low = mid1 + 1;
			high = mid2 - 1;
			res = min(res, t1);
		}
	}
	return res;

}
int main() {
#ifndef ONLINE_JUDGE
	freopen("/Users/seeva92/Workspace/Contests/1.txt", "r", stdin);
	freopen("/Users/seeva92/Workspace/Contests/2.txt", "w", stdout);
#endif
	ios::sync_with_stdio(false);
	cin.tie(0);
	// cout << setprecision(5) << fixed << eps << '\n';
	cin >> t;
	int a1, b1, a2, b2;
	while (t--) {
		cin >> a1 >> b1 >> a2 >> b2 >> v1 >> v2;
		if (a1 == a2) {
			X1 = a1, Y1 = b1, X2 = a2, Y2 = b2;
			cout << setprecision(5) << fixed << (sqrt((X1 - X1) * (X1 - X1) + (0 - Y1) * (0 - Y1)) / v1) + (sqrt((X2 - X2) * (X2 - X2) + (0 - Y2) * (0 - Y2)) / v2);
		}
		else if (a1 > a2) {
			X1 = a2, Y1 = b2, X2 = a1, Y2 = b1; swap(v1, v2);
			cout << setprecision(5) << fixed << findMin(X1, X2);
		}
		else {
			X1 = a1, Y1 = b1, X2 = a2, Y2 = b2;
			cout << setprecision(5) << fixed << findMin(X1, X2);
		}
		cout << '\n';
	}
}

// X1 = -1000000000, Y1 = -20, X2 = 1000000000, Y2 = 20;
// v1 = 40, v2 = 5;
// Y12 = -Y1 * -Y1;
// Y22 = -Y2 * -Y2;
// cout << fixed << eps << '\n';
// cout << findMin(3, 10);
// for ( double i = X1; i <= X2; i = i + .10) {

// cout << i << " " << (sqrt((X1 - i) * (X1 - i) + (0 - Y1) * (0 - Y1)) / v1) + (sqrt((X2 - i) * (X2 - i) + (0 - Y2) * (0 - Y2)) / v2) << '\n';
// }

File: HackerEarth/Sep/Sep_Circuits/Rescuer.py
import sys
from math import sqrt
sys.stdin = open("/Users/seeva92/Workspace/Contests/1.txt", "r")
sys.stdout = open("/Users/seeva92/Workspace/Contests/2.txt", "w")
t,X1,X2,Y1,Y2,v1,v2 = 0,0.0,0.0,0.0,0.0,0.0,0.0
eps = 1e-8
def findMin(low,high):
	res = 0.0;
	# print(X1,X2,Y1,Y2,V1,V2)
	while(low<=high):
		mid = low + ((high - low) / 2.0);
		midv = (sqrt((X1 - mid) * (X1 - mid) + ( -Y1) * ( -Y1)) / V1) + (sqrt((X2 - mid) * (X2 - mid) + ( -Y2) * ( -Y2)) / V2);
		midpv = (sqrt((X1 - (mid - eps)) * (X1 - (mid - eps)) + ( -Y1) * ( -Y1)) / V1) + (sqrt((X2 - (mid - eps)) * (X2 - (mid - eps)) + ( -Y2) * ( -Y2)) / V2);
		midnv = (sqrt((X1 - (mid + eps)) * (X1 - (mid + eps)) + ( -Y1) * ( -Y1)) / V1) + (sqrt((X2 - (mid + eps)) * (X2 - (mid + eps)) + ( -Y2) * (-Y2)) / V2);
		if(midpv >= midv and midv<= midnv):
			return midv;
		if(midpv >= midv and midv >= midnv):
			low = mid+eps
			res = midnv
		else:
			res = midpv
			high = mid-eps
	return res

t = int(input())
for ti in range(t):
	X1,Y1,X2,Y2,V1,V2 = map(float,input().split());
	if(X1 == X2):
		print("{0:.5f}".format(sqrt((X1-X1)*(X1-X1) + (-Y1)*(-Y1))/V1 + sqrt((X2-X2)*(X2-X2) + (-Y2)*(-Y2))/V2))
	elif X1>X2:
		X1,X2,Y1,Y2,V1,V2 = X2,X1,Y2,Y1,V2,V1		
		print("{0:.5f}".format(findMin(X1,X2)))
	else:
		print("{0:.5f}".format(findMin(X1,X2)))


# 208.88170644163475
# 140644.7312710823
# 26185.941929229353
# 210.2744743452438
# 139.26176693442022
# 2.3570226039551585

# 208.71999
# 140613.71400
# 26185.93485
# 37.18089
# 139.11210
# 2.31265


